src/Forms/GridField/GridFieldDetailForm.php

```diff
@@ -113,6 +122,17 @@ public function handleItem($gridField, $request)
         $requestHandler = $gridField->getForm()->getController();
         $record = $this->getRecordFromRequest($gridField, $request);
         if (!$record) {
+            // Look for the record elsewhere in the CMS
+            $redirectDest = $this->getLostRecordRedirection($gridField, $request);
+            // Don't allow infinite redirections
```

Comment:
How is this avoiding infinite redirections?

Reply:
Harumph, looks like I lost the relevant code. Will fix.

Reply:
@sminnee Still keen to fix this?